### PR TITLE
Support provider and Columns view-model in <uni-data-table>

### PR DESCRIPTION
**Pull request by platosha**
Key changes:

- Public view-model `<uni-data-table>` properties are combined into a single
  `provider` object to simplify binding with a view-model.
- Added `provider.Columns` array support, and a public `index` property
  in `<uni-data-table-column>`.
- The columns are generated automatically from the `provider.Columns` array.
- A developer can override any generated column in HTML using explicit
  `<uni-data-table-column index="__I__">`, where `__I__` is the index
  of the column to override from the `Columns` array.

Other related changes:

- Added `<uni-data-table-sorter>` and `<uni-data-table-filter>` aliases
  of `<vaadin-grid-sorter>` and `<vaadin-grid-filter>` helper elements.
- `PageSize` is renamed to `provider.PageSize$`
- `Size` is renamed to `provider.RowsCount`
- `RowsData` is renamed to `provider.Pages`

### CODE REVIEW

File: components/uni-data-table/README.md [hunk 1]
```diff
@@ -32,15 +32,17 @@ $ bower install uniform.css --save
...
-        rows-data='[
-            {
-                "Rows": [
-                    {"FirstName": "Jane", "LastName": "Doe"},
-                    {"FirstName": "Judy", "LastName": "Roe"}
-                ]
-            }
-        ]'>
+    <uni-data-table provider='{
+            "RowsCount": 2,
```
Comment (warpech) [inline components/uni-data-table/README.md]: This doesn't match the property in the helper library, which is called `TotalRows`.

File: components/uni-data-table/uni-data-table.html [hunk 2]
```diff
@@ -95,50 +91,84 @@
 
         static get observers() {
             return [
-                '__rowsDataObserver(rowsData, rowsData.splices, size)'
+                '__columnsObserver(provider, provider.Columns, provider.Columns.splices)',
+                '__rowsDataObserver(provider.Pages, provider.Pages.splices, provider.RowsCount)'
```
Comment (warpech) [inline components/uni-data-table/uni-data-table.html]: `RowsCount` doesn't match the property in the helper library, which is called `TotalRows`.

Source: https://github.com/Starcounter/Starcounter.Uniform/blob/80d9dc3c135ffeef247b9c0aca597f88278ed260/Starcounter.Uniform/ViewModels/UniDataTable.json#L23

@PatrykSzwer please clarify which is correct.

File: components/uni-data-table/uni-data-table.html [hunk 3]
```diff
@@ -95,50 +91,84 @@
...
                 // If the page was loaded, invoke the callback right away
                 this.__invokePageCallback(page, callback);
             } else {
                 // Otherwise, fetch the page from the server
                 // Store the pending callback for the page given number
                 this.__callbacksByPageMap.set(page, callback);
                 // Set the trigger property to invoke the server’s handler
-                this.page = page;
-                // Now we wait for the server to add the page to the `rowsData`
+                this.provider.CurrentPage$ = page;
```
Comment (warpech) [inline components/uni-data-table/uni-data-table.html]: In the helper library, there is no `CurrentPage$` but `Pagination.CurrentPageIndex$`

Source: https://github.com/Starcounter/Starcounter.Uniform/blob/80d9dc3c135ffeef247b9c0aca597f88278ed260/Starcounter.Uniform/ViewModels/UniDataTable.json#L13

@PatrykSzwer please clarify which is correct.

File: components/uni-data-table/README.md (hunk 1 above)
Comment (PatrykSzwer) [reply components/uni-data-table/README.md]: I'm clarifying: `TotalRows` is correct.

File: components/uni-data-table/uni-data-table.html (hunk 2 above)
Comment (PatrykSzwer) [reply components/uni-data-table/uni-data-table.html]: Same as above.

File: components/uni-data-table/uni-data-table.html (hunk 3 above)
Comment (PatrykSzwer) [reply components/uni-data-table/uni-data-table.html]: I'm clarifying: `Pagination.CurrentPageIndex$` is correct.

**Comment by PatrykSzwer** [comment]
For clarification, current, up-to-date view-model can be found e.g here: https://github.com/Starcounter/Starcounter.Uniform/blob/joozek/Starcounter.Uniform/ViewModels/UniDataTable.json#L7

I'll update readme to have it.

**Comment by warpech** [comment]
Merging as `uni-data-table` now supports the agreed sorting and filtering functionality with the final view-model structure.

**Pull request merged**